From 993155c5461164f9eca73c43091865ff32906601 Mon Sep 17 00:00:00 2001
From: stickz <stickman002@mail.com>
Date: Sat, 30 Dec 2023 06:57:13 -0500
Subject: [PATCH] Fix curses ui memory leak

Resolves a potential memory leak with the curses UI when filtering torrents
---
 src/ui/download_list.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ui/download_list.cc b/src/ui/download_list.cc
index f1d6af5c6..7cb0e9a89 100644
--- a/src/ui/download_list.cc
+++ b/src/ui/download_list.cc
@@ -272,6 +272,7 @@ DownloadList::receive_view_input(Input type) {
           std::getline(ss, view_name_var, ',');
           if (current_view()->name() == rak::trim(view_name_var)) {
               control->core()->push_log_std("View '" + current_view()->name() + "' can't be filtered.");
+              delete input;
               return;
           }
       }
@@ -281,6 +282,7 @@ DownloadList::receive_view_input(Input type) {
     break;
 
   default:
+    delete input;
     throw torrent::internal_error("DownloadList::receive_view_input(...) Invalid input type.");
   }
 
